pkg/ssh/ssh.go

```diff
@@ -161,15 +174,30 @@ func (s *Client) starting(ctx context.Context) error {
 
 	retryOpts := retry.Opts{MaxBackoff: 16 * time.Second, InitialBackoff: 1 * time.Second}
 	go retry.Forever(retryOpts, func() error {
+		restartTime := time.Now()
```

Comment:
I think we can simplify this by just measuring the SSH connection time. We can just use this point as the start time, and ignore whether its a start or a restart. That should reduce the amount of things we need to pass to logsMetricsParser.